server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -431,34 +389,28 @@ public int getType()
       return type;
     }
 
-    public SegmentId getSegmentId()
+    public String getSegmentIdentifier()
     {
-      return segment.getId();
+      return segment.getId().toString();
     }
 
     public long getSegmentSize()
     {
       return segment.getSize();
     }
 
-    public void addCallbacks(Collection<LoadPeonCallback> newCallbacks)
-    {
-      synchronized (callbacks) {
-        callbacks.addAll(newCallbacks);
-      }
-    }
-
     public void addCallback(LoadPeonCallback newCallback)
     {
       synchronized (callbacks) {
         callbacks.add(newCallback);
       }
     }
 
-    public List<LoadPeonCallback> getCallbacks()
+    LoadPeonCallback[] getCallbacks()
```

Comment:
A copying method shouldn't be called "getCallbacks()", perhaps, "snapshotCallbacks()"